drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c

```diff
@@ -837,13 +851,15 @@ static void job_abort(void *priv)
 static inline unsigned int get_sizeimage(int bpl, int width, int height,
 					 struct bcm2835_codec_fmt *fmt)
 {
+	unsigned int multiplier = max(fmt->size_multiplier_x2, 1);
+
 	if (fmt->flags & V4L2_FMT_FLAG_COMPRESSED) {
 		if (width * height > 1280 * 720)
-			return DEF_COMP_BUF_SIZE_GREATER_720P;
+			return DEF_COMP_BUF_SIZE_GREATER_720P * multiplier;
 		else
-			return DEF_COMP_BUF_SIZE_720P_OR_LESS;
+			return DEF_COMP_BUF_SIZE_720P_OR_LESS * multiplier;
```

Comment:
OK, wasn't used for compressed formats. 

If you need a bigger buffer for JPEGs, then better to special case them in here rather than fudging all compressed formats.
```
if (fmt->flags & V4L2_FMT_FLAG_COMPRESSED) {
	if (fmt->fourcc == V4L2_PIX_FMT_JPEG) {
		return XXXX;
	} else {
		if (width * height > 1280 * 720)
			return DEF_COMP_BUF_SIZE_GREATER_720P;
		else
			return DEF_COMP_BUF_SIZE_720P_OR_LESS;
	}
} else {
...
````
I though JPEG always came in at less than unity.

Reply:
Okay! A JPEG image can be very large with some parameters.